### ISSUE TITLE
Sort files by: Same as folder

### ISSUE THREAD

**Issue by Auxdoozy** [jurplel/qView#667]
Windows photo viewer displays images in the same order as they are in the folder. Is there anyway for qView to do the same?

**Comment by Auxdoozy** [comment]
This is because some folders are sorted by date while others are sorted by name, so being able to scroll through the images of each folder in the order that they are presented within that folder would be better.

**Comment by Auxdoozy** [comment]
This is because some folders are sorted by date while others are sorted by name, so being able to scroll through images in the order that they are presented within each folder would be ideal.

**Comment by Auxdoozy** [comment]
This is because some folders are sorted by date while others are sorted by name, so being able to scroll through images in the order that they are presented within each specific folder would be ideal.

**Comment by Auxdoozy** [comment]
This is because some folders are sorted by date while others are sorted by name, so being able to scroll through images in the same order that they are presented within each specific folder would be ideal.

**Comment by Auxdoozy** [comment]
Any update on this?

**Comment by Auxdoozy** [comment]
> Right click in the image view qView, Tools>Settings>Miscellaneous>Sort files by. And select the desired item

I'm not sure you understand my request. The issue is some folders are sorted by name while others by date. So to accurately browse each folders contents you would need to change the settings in qView each time, which obviously isn't an ideal solution. Instead I'm requesting an option for qView to automatically sort by the same order that is in the folder, the same way Windows Photo Viewer does.

**Comment by Auxdoozy** [comment]
> Right click in the image view qView, Tools>Settings>Miscellaneous>Sort files by. And select the desired item

I'm not sure you understand my request. The issue is some folders are sorted by name while others by date. So to accurately browse each folders contents you would need to change the settings in qView each time, which obviously isn't an ideal solution. Instead I'm requesting an option for qView to automatically sort by the same order that is in the folder.

**Comment by jurplel** [comment]
This suggestion makes sense, but it would definitely require extensive platform-specific code and we might not be able to make it work in all cases (e.g. linux without KDE or Gnome)

**Comment by Auxdoozy** [comment]
> This suggestion makes sense, but it would definitely require extensive platform-specific code and we might not be able to make it work in all cases (e.g. linux without KDE or Gnome)

Fair enough, thanks for the reply. In that case is there anyway you could apply it just for the windows version? As a long time Windows photo viewer user this is the only feature that I've struggled with since making the switch, apart from that qview is brilliant and better in every way!

**Comment by jurplel** [comment]
> > This suggestion makes sense, but it would definitely require extensive platform-specific code and we might not be able to make it work in all cases (e.g. linux without KDE or Gnome)
> 
> Fair enough, thanks for the reply. In that case is there anyway you could apply it just for the windows version? As a long time Windows photo viewer user this is the only feature that I've struggled with since making the switch, apart from that qview is brilliant and better in every way! 

I would definitely consider it but there is a long backlog of stuff I need to get to + I have basically no free time at the moment. Maybe in the future!

**Comment by 19X17** [comment]
Just some thought: if the app can remember the sort order on a per folder basis It could be easier to be implemented cross-platform? I am using macOS.

**Comment by conceptualspace** [comment]
agreed; a common usecase is to open a directory with images via the file manager, view one in qview and proceed to advance to the next image which may not be the expected next image depending on the sort mode of the directory.

im not sure about other OS but at least in gnome on linux we have file system metadata stored with GVFS; a directory's sort order can be determined with `gio info <directory>`. are you open to a pull request to add a "per folder default" sort option, at least for gnome users?
